lib/NetPacksLib.cpp

```diff
@@ -553,15 +549,15 @@ void TryMoveHero::applyGs(CGameState *gs)
 	{
 		const TerrainTile &tt = gs->map->getTile(h->convertToVisitablePos(end));
 		assert(tt.visitableObjects.size() >= 1  &&  tt.visitableObjects.back()->ID == Obj::BOAT); //the only visitable object at destination is Boat
-		CGBoat *boat = static_cast<CGBoat*>(tt.visitableObjects.back());
+		auto * boat = dynamic_cast<CGBoat *>(tt.visitableObjects.back());

```

Comment:
Here also.

Reply:
Answered earlier. You think than is should remain static_cast with possibility of failure?

Reply:
```suggestion
		auto * boat = dynamic_cast<CGBoat *>(tt.visitableObjects.back());
		assert(boat);
```